packages/vincent-tools/vincent-tool-uniswap-swap/src/lib/lit-actions/utils/estimate-gas.ts

```diff
@@ -0,0 +1,42 @@
+import { ethers } from "ethers";
+
+export interface GasParams {
+    estimatedGas: ethers.BigNumber;
+    maxFeePerGas: ethers.BigNumber;
+    maxPriorityFeePerGas: ethers.BigNumber;
+}
+
+export const getGasParams = async (
+    provider: ethers.providers.Provider,
+    block: ethers.providers.Block,
+    feeData: ethers.providers.FeeData,
+    estimatedGas: ethers.BigNumber
+): Promise<GasParams> => {
+    // Determine baseFeePerGas, with multiple fallbacks
+    let baseFeePerGas = block.baseFeePerGas;
+    if (baseFeePerGas == null) {
+        // Fallback to feeData.gasPrice if available
+        baseFeePerGas = feeData && feeData.gasPrice ? feeData.gasPrice : await provider.getGasPrice();
```

Comment:
I think the `provider.getGasPrice` will put us over the 50 HTTP request limit for LAs, but I when I tried testing to confirm (by making the call at the start of the function), I got a weird issue with not being able to resolve PKP token info - not sure how it's related, but going to continue to investigate

I've pinged Chris to up the HTTP limit for LA, so making this additional request here shouldn't be an issue afterwards

Reply:
Chris said he'd be bumping the limit to 75